packages/palette/src/elements/Modal/index.tsx

```diff
@@ -0,0 +1,217 @@
+import React, { SFC, useEffect, useRef, useState } from "react"
+import { animated, useSpring } from "react-spring"
+import styled from "styled-components"
+import { color, space } from "../../helpers"
+import { CloseIcon } from "../../svgs"
+import { ArtsyLogoBlackIcon } from "../../svgs/ArtsyLogoBlackIcon"
+import { Box } from "../Box"
+import { Flex } from "../Flex"
+import { Spacer } from "../Spacer"
+import { Serif } from "../Typography"
+
+interface ModalProps {
+  FixedButton?: JSX.Element
+  hasLogo?: boolean
+  isWide?: boolean
+  onClose?: () => void
+  show?: boolean
+  title?: string
+}
+
+interface TransitionElementProps {
+  isWide?: boolean
+  show?: boolean
+}
+
+const usePrevious = value => {
+  const ref = useRef(null)
+  useEffect(() => {
+    ref.current = value
+  }, [value])
+  return ref.current
+}
```

Comment:
Can this be moved to our [utils directory](https://github.com/artsy/palette/tree/master/packages/palette/src/helpers)?

Reply:
Sure just moved it